File: kt/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/syntax/CompilationUnitSyntax.kt
package dev.phytolizer.minsk.analysis.syntax

class CompilationUnitSyntax(val statement: StatementSyntax, val endOfFileToken: SyntaxToken) : SyntaxNode() {
    override val kind: SyntaxKind
        get() = SyntaxKind.CompilationUnit
    override val children: List<SyntaxNode>
        get() = listOf(statement, endOfFileToken)
}

File: cs/Minsk/CodeAnalysis/Syntax/CompilationUnitSyntax.cs
namespace Minsk.CodeAnalysis.Syntax;

public class CompilationUnitSyntax : SyntaxNode
{
    public CompilationUnitSyntax(StatementSyntax statement, SyntaxToken endOfFileToken)
    {
        Statement = statement;
        EndOfFileToken = endOfFileToken;
    }

    public StatementSyntax Statement { get; }
    public SyntaxToken EndOfFileToken { get; }
    public override SyntaxKind Kind => SyntaxKind.CompilationUnit;
    public override IEnumerable<SyntaxNode> Children => new SyntaxNode[] { Statement, EndOfFileToken };
}
